pkg/template/api/types.go

```diff
@@ -16,6 +16,10 @@ type Template struct {
 	// Optional: Parameters is an array of Parameters used during the
 	// Template to Config transformation.
 	Parameters []Parameter `json:"parameters,omitempty"`
+
+	// Optional: ObjectLabels is a set of labels that are applied to every
+	// object during the Template to Config transformation
+	ObjectLabels map[string]string `json:"objectlabels,omitempty"`
```

Comment:
Also, none of these objects should really be in the root of the template - a template should have a section that is specifically for the items, and one specifically for the things to change.

It should really be something like:

```
{
  "metadata": { // meta about the template },
  "objects": [] // not "items", because then Template is interpreted by generic behavior to be a List
  "spec": {
    "parameters": [], // environment substitution
    "label": {
      "operation": "Add",
      "labels":    {} // these get added
    }
  }
}
```

We don't have to do that in v1beta1, but for v1beta2 it should be easy to clear the "spec" (or whatever it should be called) in one operation.

----- Original Message -----
> > @@ -16,6 +16,10 @@ type Template struct {
> >  	// Optional: Parameters is an array of Parameters used during the
> >  	// Template to Config transformation.
> >  	Parameters []Parameter `json:"parameters,omitempty"`
> > +
> > +	// Optional: ObjectLabels is a set of labels that are applied to every
> > +	// object during the Template to Config transformation
> > +	ObjectLabels map[string]string `json:"objectlabels,omitempty"`
> 
> @mfojtik it's ok with me, but i think @smarterclayton is going to reply with
> "Labels" again :)
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/982/files#r24756230